query/sources/impl/src/main/java/org/codice/ddf/admin/sources/commons/utils/OpenSearchSourceUtils.java

```diff
@@ -0,0 +1,162 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.sources.commons.utils;
+
+import static org.codice.ddf.admin.common.message.DefaultMessages.unknownEndpointError;
+import static org.codice.ddf.admin.sources.commons.SourceUtilCommons.SOURCES_NAMESPACE_CONTEXT;
+import static org.codice.ddf.admin.sources.commons.SourceUtilCommons.createDocument;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import org.codice.ddf.admin.common.Report;
+import org.codice.ddf.admin.common.ReportWithResult;
+import org.codice.ddf.admin.common.fields.common.CredentialsField;
+import org.codice.ddf.admin.common.fields.common.HostField;
+import org.codice.ddf.admin.common.fields.common.UrlField;
+import org.codice.ddf.admin.sources.fields.type.OpenSearchSourceConfigurationField;
+import org.codice.ddf.admin.sources.fields.type.SourceConfigUnionField;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+public class OpenSearchSourceUtils {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OpenSearchSourceUtils.class);
+
+    private static final List<String> URL_FORMATS = ImmutableList.of(
+            "https://%s:%d/services/catalog/query",
+            "https://%s:%d/catalog/query");
+
+    public static final Map<String, String> GET_CAPABILITIES_PARAMS = ImmutableMap.of("q",
+            "test",
+            "mr",
+            "1",
+            "src",
+            "local");
+
+    private static final String TOTAL_RESULTS_XPATH = "//os:totalResults|//opensearch:totalResults";
+
+    private final RequestUtils requestUtils;
+
+    public OpenSearchSourceUtils() {
+        this(new RequestUtils());
+    }
+
+    public OpenSearchSourceUtils(RequestUtils requestUtils) {
+        this.requestUtils = requestUtils;
+    }
+
+    /**
+     * Attempts to create an OpenSearch configuration with the provided URL and credentials. If a configuration
+     * is not found or created, the {@link ReportWithResult}'s will have errors.
+     *
+     * @param urlField The URL to probe for OpenSearch capabilities
+     * @param creds    optional credentials to send with Basic Auth header
+     * @return a {@link ReportWithResult} containing the {@link SourceConfigUnionField} or containing {@link org.codice.ddf.admin.common.message.ErrorMessage}s on failure.
+     */
+    public ReportWithResult<SourceConfigUnionField> getOpenSearchConfig(UrlField urlField, CredentialsField creds) {
+        Report errors = verifyOpenSearchCapabilities(urlField, creds);
+        ReportWithResult<SourceConfigUnionField> configResult = new ReportWithResult<>();
+        if (errors.containsErrorMsgs()) {
```

Comment:
We should call `configResult.addMessages(errors)` and check `configResult` for errors instead. This way warning's will be included from `verifyOpenSearchCapabilities` if they ever exist.